compiler/lib/flow.ml

```diff
@@ -392,6 +397,29 @@ let direct_approx (info : Info.t) x =
         y
   | _ -> None
 
+let rec the_shape_of info x =
+  get_approx
+    info
+    (fun x ->
+      if Var.ISet.mem info.info_possibly_mutable x
+      then Shape.Bot "possibly_mutable"
+      else
+        match info.info_defs.(Var.idx x) with
+        | Expr (Block (_, a, _, true)) ->
```

Comment:
I think that could be the following, though it probably does not make the result any better: since the block escapes, we know that it is not possibly mutable only since it is immutable.
```suggestion
        | Expr (Block (_, a, _, _)) when not (Var.ISet.mem info.info_possibly_mutable x) ->
```

Reply:
It does improve the result

Reply:
see https://github.com/ocsigen/js_of_ocaml/pull/1594/commits/c7c009b1198bb85658300a87057ba25e818b44b4

Reply:
It no longer improve the result after reverting the change in `block_escape`

Reply:
Indeed, it seemed strange to me that it would make any difference.